packages/core/e2e/product-channel.e2e-spec.ts

```diff
@@ -0,0 +1,390 @@
+        //         input: {
+        //             productIds: [product1.id],
+        //             channelId: 'T_2',
+        //         },
+        //     });
+
+        //     expect(removeProductsFromChannel[0].channels.map(c => c.id)).toEqual(['T_1']);
+        // });
+    });
+
+    describe('assigning ProductVariant to Channels', () => {
+        let product1: GetProductWithVariants.Product;
+
+        beforeAll(async () => {
+            await adminClient.asSuperAdmin();
+            await adminClient.setChannelToken(E2E_DEFAULT_CHANNEL_TOKEN);
+
+            const { product } = await adminClient.query<
+                GetProductWithVariants.Query,
+                GetProductWithVariants.Variables
+            >(GET_PRODUCT_WITH_VARIANTS, {
+                id: 'T_2',
+            });
+            product1 = product!;
+        });
+
+        it(
+            'throws if attempting to assign ProductVariant to channel to which the admin has no access',
+            assertThrowsWithMessage(async () => {
+                await adminClient.asUserWithCredentials('admin2@test.com', 'test');
+                await adminClient.query<
+                    AssignProductVariantsToChannel.Mutation,
+                    AssignProductVariantsToChannel.Variables
+                >(ASSIGN_PRODUCTVARIANT_TO_CHANNEL, {
+                    input: {
+                        channelId: 'T_3',
+                        productVariantIds: [product1.variants[0].id],
+                    },
+                });
+            }, 'You are not currently authorized to perform this action'),
+        );
+
+        it('assigns ProductVariant to Channel and applies price factor', async () => {
+            const PRICE_FACTOR = 0.5;
+            await adminClient.asSuperAdmin();
+            await adminClient.setChannelToken(E2E_DEFAULT_CHANNEL_TOKEN);
+            const { assignProductVariantsToChannel } = await adminClient.query<
+                AssignProductVariantsToChannel.Mutation,
+                AssignProductVariantsToChannel.Variables
+            >(ASSIGN_PRODUCTVARIANT_TO_CHANNEL, {
+                input: {
+                    channelId: 'T_3',
+                    productVariantIds: [product1.variants[0].id],
+                    priceFactor: PRICE_FACTOR,
+                },
+            });
+
+            expect(assignProductVariantsToChannel[0].channels.map(c => c.id).sort()).toEqual(['T_1', 'T_3']);
+            await adminClient.setChannelToken(THIRD_CHANNEL_TOKEN);
+            const { product } = await adminClient.query<
+                GetProductWithVariants.Query,
+                GetProductWithVariants.Variables
+            >(GET_PRODUCT_WITH_VARIANTS, {
+                id: product1.id,
+            });
+            expect(product!.channels.map(c => c.id).sort()).toEqual(['T_1', 'T_3']);
+            expect(product!.variants.map(v => v.price)).toEqual(
+                product1.variants.map(v => v.price * PRICE_FACTOR),
```

Comment:
This test is failing because you are only assigning the 1st ProductVariant to Channel T_3, wheras this test assumes all 2 variants are being returned. This is a passing assertion:

```TypeScript
 expect(product!.variants.map(v => v.price)).toEqual([product1.variants[0].price * PRICE_FACTOR]);
```